### DISCUSSION TITLE
Disable OSC 7 for new tab

### DISCUSSION THREAD

**Discussion by vasti** [wezterm/wezterm#6136] [Q&A]
### What Operating System(s) are you running on?

Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

XFCE 4

### WezTerm version

20240203-110809-5046fc22

### Ask your question!

Is there a way to disable OSC 7 so that new tabs are not started in the same directory as current tab, but in home dir?
Preferably without changing/breaking `$PS1`.

i.e. disable what is described as follows:
> When the current working directory has been set via OSC 7, spawning a new tab will use the current working directory of the current tab, so that you don't have to manually change the directory.

under [OSC 7 Escape sequence to set the working directory](https://wezfurlong.org/wezterm/shell-integration.html#osc-7-escape-sequence-to-set-the-working-directory)

**Comment by loops** [comment]
Hi Vasti,

Not sure if this is what you're looking for, but you can always give the CWD if you explicitly give
provide which command to launch in the new tab.  For example with a keys entry like:

		{
			key = "T",
			mods = "SHIFT|CTRL",
			action = wezterm.action.SpawnCommandInNewTab({ args = { "bash" }, cwd = "~" }),
		},

And the directory of the current pane won't matter at all.

**Reply by wez** [reply]
Note that `cwd = "~"` should probably be `cwd = wezterm.home_dir` if you want to reference the home directory; tilde-expansion is something performed by the shell, and wezterm isn't a shell.
